### PR TITLE
Allow using labels for nodeSelector

### PR DESCRIPTION
**Pull request by AdiZav**
Allow using labels for nodeSelector.
This then allows deploying resources to several nodes

### CODE REVIEW

File: ocp_resources/node_network_configuration_policy.py [hunk 1]
```diff
@@ -102,16 +104,20 @@ def __init__(
...
+            return list(Node.get(dyn_client=self.client, name=self.node_selector))
+        if self.node_selector_labels:
+            label_key, label_value = list(self.node_selector_labels.items())[0]
+            nodes = list(
+                Node.get(
+                    dyn_client=self.client, label_selector=f"{label_key}={label_value}"
+                )
+            )
+            return nodes
+        return None
```
Comment (myakove) [inline ocp_resources/node_network_configuration_policy.py:120]: No need, this is the default.

File: ocp_resources/node_network_configuration_policy.py [hunk 2]
```diff
@@ -102,16 +104,20 @@ def __init__(
...
-            self.nodes = list(Node.get(dyn_client=self.client, name=self.node_selector))
-        else:
-            self._node_selector = {
-                f"node-role.{self.ApiGroup.KUBERNETES_IO}/worker": ""
-            }
-            self.nodes = list(Node.get(dyn_client=self.client))
+            return list(Node.get(dyn_client=self.client, name=self.node_selector))
+        if self.node_selector_labels:
+            label_key, label_value = list(self.node_selector_labels.items())[0]
+            nodes = list(
```
Comment (myakove) [inline ocp_resources/node_network_configuration_policy.py:114]: return list(.......)

File: ocp_resources/resource.py [hunk 3]
```diff
@@ -385,6 +387,16 @@ def __init__(
...
+        self.node_selector = node_selector
+        self.node_selector_labels = node_selector_labels
+        self._node_selector = self._set_node_selector()
+
+    def _set_node_selector(self):
+        if self.node_selector:
+            return {f"{self.ApiGroup.KUBERNETES_IO}/hostname": self.node_selector}
+        if self.node_selector_labels:
+            return self.node_selector_labels
+        return None
```
Comment (myakove) [inline ocp_resources/resource.py:399]: ditto

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocp_resources/node_network_configuration_policy.py (hunk 1 above)
Comment (AdiZav) [reply ocp_resources/node_network_configuration_policy.py:120]: ACK

File: ocp_resources/node_network_configuration_policy.py (hunk 2 above)
Comment (AdiZav) [reply ocp_resources/node_network_configuration_policy.py:114]: ACK

File: ocp_resources/resource.py (hunk 3 above)
Comment (AdiZav) [reply ocp_resources/resource.py:399]: ACK

File: ocp_resources/node_network_configuration_policy.py [hunk 4]
```diff
@@ -102,16 +104,18 @@ def __init__(
...
-            self._node_selector = {
-                f"node-role.{self.ApiGroup.KUBERNETES_IO}/worker": ""
-            }
-            self.nodes = list(Node.get(dyn_client=self.client))
+            return list(Node.get(dyn_client=self.client, name=self.node_selector))
+        if self.node_selector_labels:
+            label_key, label_value = list(self.node_selector_labels.items())[0]
+            return list(
+                Node.get(
+                    dyn_client=self.client, label_selector=f"{label_key}={label_value}"
```
Comment (rnetser) [inline ocp_resources/node_network_configuration_policy.py:116]: why not support multiple labels? 
",".join(["f{label_key}={label_value}" for label_key, label_value in self.node_selector_labels.items()])

File: ocp_resources/resource.py [hunk 5]
```diff
@@ -385,6 +387,15 @@ def __init__(
         self.teardown = teardown
         self.timeout = timeout
         self.delete_timeout = delete_timeout
+        self.node_selector = node_selector
+        self.node_selector_labels = node_selector_labels
+        self._node_selector = self._set_node_selector()
```
Comment (rnetser) [inline ocp_resources/resource.py:392]: why do we need both `self.node_selector` and `self._node_selector` ? this is confusing

**Review by rnetser — CHANGES_REQUESTED** [review]

File: ocp_resources/node_network_configuration_policy.py (hunk 4 above)
Comment (myakove) [reply ocp_resources/node_network_configuration_policy.py:116]: +1

File: ocp_resources/resource.py (hunk 5 above)
Comment (myakove) [reply ocp_resources/resource.py:392]: self.node_selector is the one that we got from the use.
self._node_selector is what we set in the body {f"{self.ApiGroup.KUBERNETES_IO}/hostname": self.node_selector}

Agree, needs to be renamed

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocp_resources/resource.py [hunk 6]
```diff
@@ -385,6 +387,15 @@ def __init__(
         self.teardown = teardown
         self.timeout = timeout
         self.delete_timeout = delete_timeout
+        self.node_selector = node_selector
+        self.node_selector_labels = node_selector_labels
+        self.body_node_selector = self._set_node_selector()
```
Comment (rnetser) [inline ocp_resources/resource.py:392]: Can you please rename to something which indicates that this applies to the spec, for example `self.spec_node_selector`

**Review by rnetser — CHANGES_REQUESTED** [review]

File: ocp_resources/resource.py (hunk 6 above)
Comment (myakove) [inline ocp_resources/resource.py:392]: do you mind changing it to 
node_selector_body

File: ocp_resources/resource.py [hunk 7]
```diff
@@ -385,6 +387,15 @@ def __init__(
...
+
+    def _set_node_selector(self):
```
Comment (myakove) [inline ocp_resources/resource.py:394]: This not set anything, please rename to prepare maybe? And use body in the name to have correlation between the two.

**Review by myakove — CHANGES_REQUESTED** [review]

**Review by myakove — APPROVED** [review]

**Review by myakove — APPROVED** [review]

File: ocp_resources/catalog_source.py [hunk 8]
```diff
@@ -21,6 +21,7 @@ def __init__(
         teardown=True,
         yaml_file=None,
         delete_timeout=TIMEOUT_4MINUTES,
+        **kwargs,
```
Comment (rnetser) [inline ocp_resources/catalog_source.py:24]: please rebase

File: ocp_resources/resource.py [hunk 9]
```diff
@@ -385,6 +388,16 @@ def __init__(
         self.teardown = teardown
         self.timeout = timeout
         self.delete_timeout = delete_timeout
+        self.dry_run = dry_run
+        self.node_selector = node_selector
+        self.node_selector_labels = node_selector_labels
+        self.node_selector_spec = self._node_selector_spec_prepare()
+
+    def _node_selector_spec_prepare(self):
```
Comment (rnetser) [inline ocp_resources/resource.py:396]: please rename, have the verb first:
_prepare_node_selector_spec

**Review by rnetser — CHANGES_REQUESTED** [review]

File: ocp_resources/catalog_source.py (hunk 8 above)
Comment (AdiZav) [reply ocp_resources/catalog_source.py:24]: It's already rebased

File: ocp_resources/resource.py (hunk 9 above)
Comment (AdiZav) [reply ocp_resources/resource.py]: ack

File: ocp_resources/resource.py [hunk 10]
```diff
@@ -391,9 +391,9 @@ def __init__(
         self.dry_run = dry_run
         self.node_selector = node_selector
         self.node_selector_labels = node_selector_labels
-        self.node_selector_spec = self._node_selector_spec_prepare()
+        self.node_selector_spec = self._prepare_selector_spec()
 
-    def _node_selector_spec_prepare(self):
+    def _prepare_selector_spec(self):
```
Comment (myakove) [inline ocp_resources/resource.py:396]: I hope the last one :)
Why did you remove the work node from the name?

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocp_resources/resource.py (hunk 10 above)
Comment (AdiZav) [reply ocp_resources/resource.py:396]: I don't have any nice words to describe this mistake

**Review by rnetser — APPROVED** [review]

**Review by myakove — APPROVED** [review]
